sentry-spring/src/main/java/io/sentry/spring/SentrySpringFilter.java

```diff
@@ -0,0 +1,100 @@
+package io.sentry.spring;
+
+import static io.sentry.SentryOptions.RequestSize.*;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.Breadcrumb;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions.RequestSize;
+import io.sentry.util.Objects;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.MediaType;
+import org.springframework.util.MimeType;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Open
+public class SentrySpringFilter extends OncePerRequestFilter {
+  private final @NotNull IHub hub;
+  private final @NotNull SentryRequestResolver requestResolver;
+
+  public SentrySpringFilter(@NotNull IHub hub, @NotNull SentryRequestResolver requestResolver) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.requestResolver = Objects.requireNonNull(requestResolver, "requestResolver is required");
+  }
+
+  public SentrySpringFilter(final @NotNull IHub hub) {
+    this(hub, new SentryRequestResolver(hub));
+  }
+
+  public SentrySpringFilter() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final @NotNull HttpServletRequest servletRequest,
+      final @NotNull HttpServletResponse response,
+      final @NotNull FilterChain filterChain)
+      throws ServletException, IOException {
+    if (hub.isEnabled()) {
+      final HttpServletRequest request = resolveHttpServletRequest(servletRequest);
+      try {
+        hub.pushScope();
+        hub.addBreadcrumb(Breadcrumb.http(request.getRequestURI(), request.getMethod()));
+        hub.configureScope(
+            scope -> {
+              scope.setRequest(requestResolver.resolveSentryRequest(request));
```

Comment:
Here we're eagerly reading the request bytes into the scope which won't really be used unless we have an event.

Another alternative here is to add an event processor that holds the request (we have on the line below?) and we then only add the request information if needed and spare all the allocations.

Buffering the request is required anyway, since we need to be able to rewind the stream in case of an error.

Reply:
One issue I see is that we're passing `request` here regardless of the outcome of `resolveHttpServletRequest`. Meaning, it could have returned the normal rquest without buffering it (so regardless of the result of `qualifiesForCaching`.

I think we need to flow the result of `qualifiesForCaching` to guard calling `setRequest` to add this altogether.

Reply:
Updated. Now the request body is not set on the scope at all, and added in event processor.